bin/functional-tests/test_config.py

```diff
@@ -152,12 +152,11 @@ def test_prometheus_config_reloader_works(prometheus, kube_client):
         print(f"Exception when calling CoreV1Api->patch_namespaced_config_map: {e}\n")
 
     # This can take more than a minute.
-    for i in range(12):
+    for _ in range(12):
```

Comment:
i'd rather see a polling library or helper function than loops with sleep in them. if we hang on line 156, the loop could potentially last much longer than 2 minutes

Reply:
Good catch, we can use pytest.flaky on this.

Reply:
actually i don't think pytest flaky is the right approach here...

Reply:
https://github.com/justiniso/polling

Reply:
yes seems to be useful - on next phase lets optimise our test cases